sys/include/vfs_util.h

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (C) 2021 ML!PA Consulting GmbH
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @defgroup    sys_vfs_util    VFS helper functions
+ * @ingroup     sys_vfs
+ * @{
+ *
+ * @file
+ * @brief       VFS helper functions
+ *
+ * @author      Benjamin Valentin <benjamin.valentin@ml-pa.com>
+ */
+
+#ifndef VFS_UTIL_H
+#define VFS_UTIL_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   Writes the content of a buffer to a file
+ *          If the file already exists, it will be overwritten.
+ *
+ * @param[in]  file     Destination file path
+ * @param[in]  buf      Source buffer
+ * @param[in]  len      Buffer size
+ *
+ * @return  0 on success
+ * @return  negative error
+ */
+int vfs_file_from_buffer(const char *file, const void *buf, size_t len);
+
+/**
+ * @brief   Reads the content of a file to a buffer
+ *
+ * @param[in]  file     Source file path
+ * @param[out] buf      Destination buffer
+ * @param[in]  len      Buffer size
+ *
+ * @return  number of bytes read on success
+ * @return  negative error
```

Comment:
Could add:

> If the file is longer than the buffer, the buffer is filled and the read is reported as successful; to be sure that the file was read completely, allocate buffer that is larger than the expected size.

Reply:
Now I wonder if it would make sense to report an error if there are more bytes in the file.
Then again I could also see this function as handy if one only wants to read the header from a file.

Reply:
Either is fine as long as it's documented. My suggestion was based on what's implemented, but I would find an error (-ENOSPC?) more intuitive from a 2022 perspective. (I vaguely remember that historically there has been a preference for truncating in similar situations, but historically there have also been a lot of bugs ;-) ).

Reply:
How about

```C
int vfs_file_to_buffer(const char* file, void* buf, size_t len, bool *more);
```

Reply:
I think that'd make rather cumbersome error checking for what is supposed to be a convenience shorthand that works with a single line of code. If someone needs the full power, they can always go the open/read route.